lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -102,9 +104,10 @@ def serializable_hash_for_collection(options)
         def serializable_hash_for_single_resource
           primary_data = primary_data_for(serializer)
           relationships = relationships_for(serializer)
-          included = included_resources(@include_tree)
+          primary_data[:relationships] = relationships if relationships.any?
           hash = { data: primary_data }
-          hash[:data][:relationships] = relationships if relationships.any?
+
+          included = included_resources(@include_tree, [primary_data])
```

Comment:
Yes, this is to ensure, preemptively, that we do not include any of the primary data in the `included` hash when building the hash for a single resource.